Source/Controller/BlobViewController.swift

```diff
@@ -42,8 +43,7 @@ class BlobViewController: ContentViewController {
     }
 
     private func update() {
-        Caches.blobs.image(for: self.blob) {
-            [weak self] _, image in
+        Caches.blobs.imageOrPlaceholder(for: self.blob) { [weak self] image in
             self?.imageView.image = image
```

Comment:
How do you feel about adding a DispatchQueue.main.async here?
Because I had to dig into the implementation of that function to check if it called the completion handler in the main thread.

Reply:
I went ahead and tried to use the new `@MainActor` annotation to make this even safer. By annotating `BlobViewController` with `@MainActor` the compiler will now generate an error if anyone tries to update one of the view controllers properties (like `imageView`) from a background Task.

Reply:
(Note that this is only guaranteed to be enforced in structured concurrency contexts, but in my tests trying to wrap the update to `imageView` in `DispatchQueue.async` did generate a compiler error. More about this [here](https://www.swiftbysundell.com/articles/the-main-actor-attribute/))